test/hooks/useResult.ts

```diff
@@ -26,10 +26,11 @@ describe("useResult", () => {
   it("should provide context", async () => {
     const testEffect = Effect.map(foo, (_) => _.value)
 
-    const { result } = await waitFor(async () => renderHook(() => useResult(testEffect)))
+    const renderResult = await waitFor(async () => renderHook(() => useResult(testEffect)))
+    const result = renderResult.result.current.result
 
-    await waitFor(() => expect(Result.isSuccess(result.current.result)).toBe(true))
-    expect(result.current.result.value).toBe(1)
+    await waitFor(() => expect(Result.isSuccess(result)).toBe(true))
+    expect(result._tag === "Success" && result.value).toBe(1)
```

Comment:
could you use `assert.deepEquals` (or the `expect` equivalent) or 2 assertions for this? 
Seeing 
```
expected: 1
actual: false
```
Would be rather confusing..

Reply:
Definitely needs improvement, there was ts errors and wanted to quickly get rid of them.